synapse/federation/sender/__init__.py

```diff
@@ -297,6 +297,8 @@ async def _send_pdu(self, pdu: EventBase, destinations: Iterable[str]) -> None:
         sent_pdus_destination_dist_total.inc(len(destinations))
         sent_pdus_destination_dist_count.inc()
 
+        assert pdu.internal_metadata.stream_ordering
```

Comment:
these additional assertions are just required by mypy, since we've changed the type of `internal_metadata.stream_ordering` from `int` to `Optional[int]`.